### DISCUSSION TITLE
Improvement suggestions for Artist Searches

### DISCUSSION THREAD

**Discussion by eric-s-weaver** [JohnnyCrazy/SpotifyAPI-NET#590] [Q&A]
We are using SpotifyAPI-Net in our Smartspeaker product.  It works great but we often get poor results with certain searches.  Here's the latest example.  A voice search like "Play songs by the Who" triggers a SearchItemsAsync with SearchType.All.  The resulting SearchItem object has "The Who" as the first item in the Artists like (GOOD), Who Albums in the Albums list (GOOD) but the Tracks list contains some Who songs and then other songs with "Who" or "Whom" or "The Whole" in the title.  So this is not idea.

If I take the top Artist in the results which is accurate and I then call GetArtistsTopTracksAsync, I only get 10 tracks returned which isn't enough.

1) Ideally, the GetArtistsTopTracksAsync would return the maximum 50 results like the other calls
2) The other possible solution is to specify an Artist ID in the SearchItemsAsync call to filter on only that ARtist.

Any other ideas?

Eric Weaver
HandsFree Health

**Comment by JohnnyCrazy** [comment]
Hi Eric,

While those are great suggestions, I'm just providing an API **wrapper** and I am not in any way responsible how spotify's backend API returns payloads. You will have to contact them if you have feedback regarding API endpoints and their results.

**Reply by eric-s-weaver** [reply]
I see.  I wasn't sure if you were limiting the # of tracks returned in any way.

**Reply by eric-s-weaver** [reply]
So I posted this question with Spotify and here's their response.  Does this make sense?  How can I do this with your API Wrapper?

Hey @eweaver, thanks for joining the conversation on the Spotify Community!

 

_Hmm, you might be able to use search operators in your Search request. This could be the query parameter for this scenario: 

artist:"The Who"
 Let me know if that helps you get on!

 

Have a great day,

Hubo_
